src/main/java/teammates/ui/template/StudentListStudentData.java

```diff
@@ -30,25 +30,26 @@ public StudentListStudentData(String googleId, String studentName, String studen
         this.sessionToken = sessionToken;
         this.courseStudentDetailsLink =
                 furnishLinkWithCourseEmailAndUserId(Const.ActionURIs.INSTRUCTOR_COURSE_STUDENT_DETAILS_PAGE,
-                                                    course, studentEmail, googleId);
+                                                    course, studentEmail, googleId, null);
         this.courseStudentEditLink =
                 furnishLinkWithCourseEmailAndUserId(Const.ActionURIs.INSTRUCTOR_COURSE_STUDENT_DETAILS_EDIT,
-                                                    course, studentEmail, googleId);
+                                                    course, studentEmail, googleId, null);
         this.courseStudentRemindLink = furnishLinkWithCourseEmailAndUserId(Const.ActionURIs.INSTRUCTOR_COURSE_REMIND,
-                                                                           course, studentEmail, googleId);
+                                                                           course, studentEmail, googleId, previousPage);
         this.courseStudentDeleteLink = furnishLinkWithCourseEmailAndUserId(Const.ActionURIs.INSTRUCTOR_COURSE_STUDENT_DELETE,
-                                                                           course, studentEmail, googleId);
+                                                                           course, studentEmail, googleId, null);
         this.courseStudentRecordsLink = furnishLinkWithCourseEmailAndUserId(Const.ActionURIs.INSTRUCTOR_STUDENT_RECORDS_PAGE,
-                                                                            course, studentEmail, googleId);
+                                                                            course, studentEmail, googleId, null);
     }
 
     private String furnishLinkWithCourseEmailAndUserId(String rawLink, String course, String studentEmail,
-                                                       String googleId) {
+                                                       String googleId, String previousPage) {
```

Comment:
Instead of modifying this method, simply modify `rawLink` before passing it here.